src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +832,46 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        private void EmbedCompilerFlags(CommonPEModuleBuilder module)
+        {
+            byte[] bytes = new byte[0];
+
+            // TODO: Where to get compiler version? 
+
+            // TODO: Where to get source encoding
+
+            _debugMetadataOpt.AddCustomDebugInformation(
+                parent: EntityHandle.ModuleDefinition,
+                kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.DeterministicCompilerFlags),
+                value: _debugMetadataOpt.GetOrAddBlob(bytes));
+        }
+
+        private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
+        {
+            var bytes = new List<byte>();
+
+            // TODO: Get metadata reference information
+            // File name: Foo.exe
+            // COFF header Timestamp field (4 bytes): 0x542d5742
+            // COFF header SizeOfImage field (4 bytes): 0x32000000
+            // MVID (Guid, 24 bytes): 0x24a44d8218894463807674caf3b1c19a
+            // Output: Foo.exe\0542d5742320000000024a44d8218894463807674caf3b1c19a
+            foreach (var metadataReference in module.CommonCompilation.ExternalReferences)
+            {
+                if (metadataReference.Display is object && !IsTooLongInternal(metadataReference.Display, PdbLengthLimit))
+                {
+                    bytes.AddRange(Encoding.UTF8.GetBytes(metadataReference.Display));
```

Comment:
>bytes [](start = 20, length = 5)

One more thing - we will need to include some flags and extern aliases defined for the reference (see MetadataReferenceProperties)

Reply:
Not sure I follow. What flags and aliases do we need and why?

Reply:
MetadataReferenceProperties has the following fields:
```
        private readonly MetadataImageKind _kind;
        private readonly ImmutableArray<string> _aliases;
        private readonly bool _embedInteropTypes;
```

Encoding the kind will allow us to distinguish between an assembly reference (`-r` on command line) and a module reference (`-addmodule` on command line).

Aliases are also needed - they define available extern aliases for the reference (`-r:foo=x.dll`, `-r:bar=x.dll` on command line).

EmbedInteropTypes will allow us to distinguish between assembly reference (`-r`) and PIA (`-l`).

Reply:
These are now included